datalad/core/local/configuration.py

```diff
@@ -0,0 +1,337 @@
+        name = res['name']
+        if kwargs.get('action', None) == 'dump' and name in cfg_defs:
+            ui_def = cfg_defs[name].get('ui', [None, {}])[1]
+            for s in (ui_def.get('title'),
+                      ui_def.get('text')):
+                if s:
+                    ui.message('\n'.join(wrap(
+                        s,
+                        initial_indent='# ',
+                        subsequent_indent='# ',
+                    )))
+
+        if kwargs.get('recursive', False):
+            have_subds = res['path'] != res['refds']
+            # we need to mark up from which dataset results are reported
+            prefix = '<ds>{}{}:'.format(
+                '/' if have_subds else '',
+                Path(res['path']).relative_to(res['refds']).as_posix()
+                if have_subds else '',
+            )
+        else:
+            prefix = ''
+
+        if kwargs.get('action', None) == 'dump':
+            ui.message('{}{}={}'.format(
+                prefix,
+                ac.color_word(name, ac.BOLD),
+                res['value'] if res['value'] is not None else '',
+            ))
+        else:
+            ui.message('{}{}'.format(
+                prefix,
+                res['value'] if res['value'] is not None else '',
+            ))
+
+
+def configuration(action, scope, specs, res_kwargs, ds=None):
+    if scope == 'global':
+        cfg = dlcfg
+    else:
+        cfg = ds.config
+
+    if action not in config_actions:
+        raise ValueError("Unsupported action '{}'".format(action))
+
+    if action == 'dump':
+        if specs:
+            lgr.warning('Provided configuration selection is ignored when dumping')
+        # dumping is querying for all known keys
+        # TODO warn that specs will be ignored, if there are any
+        specs = [(n,) for n in sorted(set(cfg_defs.keys()).union(cfg.keys()))]
+        scope = None
+
+    for spec in specs:
+        if '.' not in spec[0]:
+            yield get_status_dict(
+                res_kwargs,
+                ds=ds,
+                logger=lgr,
+                status='error',
+                message=(
+                    "Configuration key without a section: '%s'",
+                    spec[0],
+                )
+            )
+            continue
+        # TODO without get-all there is little sense in having add
+        #if action == 'add':
+        #    res = _add(cfg, scope, spec)
+        if action == 'get':
+            res = _get(cfg, scope, spec[0])
+        elif action == 'dump':
+            res = _dump(cfg, spec[0])
+        # TODO this should be there, if we want to be comprehensive
+        # however, we turned this off by default in the config manager
+        # because we hardly use it, and the handling in ConfigManager
+        # is not really well done.
+        #elif action == 'get-all':
+        #    res = _get_all(cfg, scope, spec)
+        elif action == 'set':
+            res = _set(cfg, scope, *spec)
```

Comment:
If the user does a `git config`-inspired `set a.b.c val`, this will
fail with

```
[ERROR  ] _set() missing 1 required positional argument: 'value' [configuration.py:configuration:275] (TypeError)
```

It seems worth giving a more informative error message since I think mixing the two forms will be a common mistake.

Reply:
```
% datalad configuration set mike.tyson winner
[ERROR  ] Values must be provided for all configuration settings. Missing: ['mike.tyson', 'winner'] [configuration.py:__call__:179] (ValueError) 
```